cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -1068,11 +1067,7 @@ def _add_head_node(self):
                     "instance_types_data_s3_key": f"{self.bucket.artifact_directory}/configs/instance-types-data.json",
                     "custom_node_package": self.config.custom_node_package or "",
                     "custom_awsbatchcli_package": self.config.custom_aws_batch_cli_package or "",
-                    "head_node_imds_secured": "true"
-                    if get_property_head_node_imds_secured(
-                        self.config.head_node.imds.secured, self.config.scheduling.scheduler
-                    )
-                    else "false",
+                    "head_node_imds_secured": "true" if self.config.head_node.imds.secured else "false",
```

Comment:
minor: would it make sense to instead use `str(self.config.head_node.imds.secured).lower()` here?

The reason being that if the parameter object is `None` then this defaults to not securing IMDS on the head node. I don't think that can happen as currently written, so this is a bit overly paranoid.

Reply:
I agree with you

Reply:
I agree with you: in this place we can assume that imds.secured is not None.
I agree because, otherwise, we are silently introducing another defaulting behavior, which is not desired.